### ISSUE TITLE
Give engi's a choice when upgrading SG's

### ISSUE THREAD

**Issue by squeek502** [fortressforever/fortressforever#211]
> **squeek:**
> right, i play under the assumption that level 2 is good enough
> level 3 is a luxury
> **AfterShock:**
> yeah
> level 2 is a lot better than 1
> it didn't used to be that way
> **squeek:**
> which leads to the age old thing of 'i wish upgrading was a choice'
> **AfterShock:**
> but that doesnt mean this way isnt better
> (i mean level 3 used to be more common?)
> **squeek:**
> because getting a level 2 and then having 130 cells means your sg is as good as dead
> level 3 is just as common
> i think im the only one that will maintain level 2s
> but it's a bad idea because if you get >= 130 cells then your first repair costs 130 no matter what
> **AfterShock:**
> in the old days of tfc, you'd occasionally get big metal bags from soldiers who died with a lot of armor (after you repairing them)
> **squeek:**
> which is major lame
> **AfterShock:**
> well we could make it so you can only upgrade a repaired SG
> but whacking away would then sometimes accidenrally upgrade it
> **squeek:**
> yeah the ideal would be to make it a legit choice somehow
> but not hidden away as a menu option or something
> maybe a hold to upgrade thing
> where you do something different if you hold fire
> so to just repair you'd press/release over and over
> **AfterShock:**
> hmm
> **squeek:**
> we've always had good success with charging mechanics!
> not buggy at all!

**Comment by FDA-** [comment]
I hate press-release toggles. My railgun and sniper rifle both currently operate under press-press toggles. That's just me though.

**Comment by R00Ki3** [comment]
2-3 hits on an undamaged SG to upgrade? While its damaged it always gets repaired first.

**Comment by TYRANNOSAURUSCOW** [comment]
I like the idea.  Non-buggy feeling implementation would be key though, yes - during the upgrade would the engy have to [be and then] stay a certain distance from the SG, or alternatively would he get locked in place?

As well as making repairing/upgrading a choice, which I agree would be good, it would also help stagger the upgrade process (something I've wanted more of for years for many reasons) and make the engy more vulnerable while upgrading (creating more depth in that choice).  Both good things imo and both to a degree dependent on the upgrade time, so although I assume this would probably be quite short initially (and therefore have minimal effect on these two things) could be a useful lever to think about increasing future if the concept works well.

**Comment by fpsmoto** [comment]
Agreed. Having the choice to upgrade is definitely something I'd like to see.
